spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/rsocket/RSocketWebSocketNettyRouteProvider.java

```diff
@@ -42,19 +45,35 @@
 
 	private final List<RSocketServerCustomizer> customizers;
 
-	RSocketWebSocketNettyRouteProvider(String mappingPath, SocketAcceptor socketAcceptor,
+	private final String protocols;
+
+	private final int maxFramePayloadLength;
+
+	private final boolean handlePing;
+
+	private final boolean compress;
+
+	RSocketWebSocketNettyRouteProvider(String mappingPath, Spec spec, SocketAcceptor socketAcceptor,
```

Comment:
There's no need to assign the fieds here. keeping the `Spec` object should be enough.